openspec/changes/archive/2026-07-05-extract-shared-storage-module/design.md

```
@@ -0,0 +1,174 @@
+        $c->get(Database\InteractionsTable::class),
+        Axis::User,
+    ),
+```
+
+Repos resolve `Storage::class` from the per-module container scope and remain axis-blind.
+
+### Registry snapshot/rollback preserved
+
+The current `loadItems → snapshot → mutate → write → rollback-on-failure` flow stays in both repos:
+
+```
```

markdownlint-cli2 (0.22.1)

[warning] Fenced code blocks should have a language specified

(MD040, fenced-code-language)